vertx-mongo-client/src/main/java/io/vertx/ext/mongo/MongoClient.java

```diff
@@ -225,42 +273,84 @@ static MongoClient createShared(Vertx vertx, JsonObject config) {
    * @param query  query used to match documents
    * @param resultHandler will be called when complete
    */
-  @Fluent
+  @Deprecated @Fluent
   MongoClient remove(String collection, JsonObject query, Handler<AsyncResult<Void>> resultHandler);
 
   /**
+   * Remove matching documents from a collection and return the handler with MongoClientDeleteResult result
+   *
+   * @param collection  the collection
+   * @param query  query used to match documents
+   * @param resultHandler will be called when complete
+   */
+  @Fluent
+  MongoClient removeWithMongoClientDeleteResult(String collection, JsonObject query, Handler<AsyncResult<MongoClientDeleteResult>> resultHandler);
+
+  /**
    * Remove matching documents from a collection with the specified write option
    *
    * @param collection  the collection
    * @param query  query used to match documents
    * @param writeOption  the write option to use
    * @param resultHandler will be called when complete
    */
-  @Fluent
+  @Deprecated @Fluent
   MongoClient removeWithOptions(String collection, JsonObject query, WriteOption writeOption, Handler<AsyncResult<Void>> resultHandler);
```

Comment:
No, the intention of this method is to remove matching documents from a collection correct?